crates/storage/provider/src/providers/consistent.rs

```diff
@@ -796,22 +796,26 @@ impl<N: ProviderNodeTypes> BlockReader for ConsistentProvider<N> {
         hash: B256,
         source: BlockSource,
     ) -> ProviderResult<Option<Self::Block>> {
-        match source {
-            BlockSource::Any | BlockSource::Canonical => {
-                // Note: it's fine to return the unsealed block because the caller already has
-                // the hash
-                self.get_in_memory_or_storage_by_block(
-                    hash.into(),
-                    |db_provider| db_provider.find_block_by_hash(hash, source),
-                    |block_state| Ok(Some(block_state.block_ref().recovered_block().clone_block())),
-                )
-            }
-            BlockSource::Pending => Ok(self
+        if matches!(source, BlockSource::Any | BlockSource::Pending) {
```

Comment:
I'd like to restore the previous order where we checked canonical first

and also swap the order in the matches! macro to Pending | any and canonical | any